src/awkward/operations/ak_categories.py

```diff
@@ -35,11 +35,7 @@ def action(layout, **kwargs):
             return None
 
     layout = ak.operations.to_layout(array, allow_record=False, allow_other=False)
-    layout.recursively_apply(action)
-
-    if output[0] is None:
-        return None
-    elif highlevel:
-        return ak._util.wrap(output[0], ak._util.behavior_of(array))
-    else:
-        return output[0]
+    behavior = ak._util.behavior_of(array)
+    layout.recursively_apply(action, behavior=behavior)
+
+    return ak._util.wrap(output[0], behavior, highlevel)
```

Comment:
And this is an introduction of `behavior_of` (furthering its use).

The reason it was done here is so that the `behavior` can be used in `recursively_apply`. There are a few other examples like this, in which the `behavior_of` is far from the `recursively_apply` (or `broadcast_and_apply`). This is correct, by the way: it's the merged behavior that these functions want.

(For this reason, I think the right solution to the "which internal interface should we use?" is `behavior_of`, not `like`.)

Reply:
Since it turns out that the boilerplate is sometimes necessary (because `behavior` is used in `recursively_apply`, as well as the output array), `like` is not a simplification. I was thinking that we'd eventually cull one of the interfaces, and since `behavior_of` is more general than the `like` argument, it's the `like` argument that would go.

Alternatively, the `like` argument could just get renamed as a `behavior_of` _argument_. It would still be done in two ways (explicit function call vs argument), but having the same name would clue-in code readers/maintainers that they're the same thing. Each high-level function should have exactly one appearance of "`behavior_of`", whether it's the function or the argument.